nixos/modules/services/networking/mosquitto.nix

```diff
@@ -233,15 +233,50 @@ in
         ExecStart = "${pkgs.mosquitto}/bin/mosquitto -c ${mosquittoConf}";
         ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
 
-        ProtectSystem = "strict";
-        ProtectHome = true;
+        # Hardening
+        CapabilityBoundingSet = "";
+        DevicePolicy = "closed";
+        LockPersonality = true;
+        MemoryDenyWriteExecute = true;
+        NoNewPrivileges = true;
         PrivateDevices = true;
+        PrivateUsers = true;
         PrivateTmp = true;
-        ReadWritePaths = "${cfg.dataDir}";
+        ProtectClock = true;
         ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
         ProtectKernelModules = true;
         ProtectKernelTunables = true;
-        NoNewPrivileges = true;
+        ProtectProc = "invisible";
+        ProcSubset = "pid";
+        ProtectSystem = "strict";
+        ReadWritePaths = [
+          cfg.dataDir
+          "/tmp"  # mosquitto_passwd creates files in /tmp before moving them
```

Comment:
Without this we get the following error. Adding `/tmp` here is a non-issue, because it is a `PrivateTmp` anyway.

```
server # [   11.458054] systemd[1]: mosquitto.service: Control process exited, code=exited, status=1/FAILURE
server # [   11.460356] mosquitto-pre-start[865]: + /nix/store/61qqw8pvlbik2vzrh1p4gmlcvmi0a6p2-mosquitto-2.0.10/bin/mosquitto_passwd -b /var/lib/mosquitto/passwd mqtt VERY_secret
server # [   11.463963] systemd[1]: mosquitto.service: Failed with result 'exit-code'.
server # [   11.466167] mosquitto-pre-start[870]: Error: Unable to open temporary file. Read-only file system.
```